task/nucleo-user-button/src/main.rs

```diff
@@ -67,10 +67,26 @@ pub fn main() -> ! {
     let user_leds = drv_user_leds_api::UserLeds::from(USER_LEDS.get_task_id());
     let sys = drv_stm32xx_sys_api::Sys::from(SYS.get_task_id());
 
-    let (led, rising, falling) = if let Some(config) = TASK_CONFIG {
-        (config.led, config.rising, config.falling)
+    let (led, sensitivity) = if let Some(Config {
+        rising,
+        falling,
+        led,
+    }) = TASK_CONFIG
+    {
+        let sensitivity = match (rising, falling) {
+            (true, true) => Edge::Both,
+            (false, true) => Edge::Rising,
+            (true, false) => Edge::Falling,
+            (false, false) => {
```

Comment:
(also, you could move this into the build.rs if desired)

(I realize this is a demo task and it doesn't super matter, but demo tasks do tend to get copy-pasted.)